server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/SearchAction.java

```diff
@@ -124,8 +124,8 @@ private ComponentDto getComponentByUuidOrKey(DbSession dbSession, SearchWsReques
       request.getProjectKey(),
       ComponentFinder.ParamNames.PROJECT_ID_AND_KEY);
 
-    if (!userSession.hasComponentUuidPermission(UserRole.ADMIN, component.uuid()) &&
-      !userSession.hasComponentUuidPermission(UserRole.USER, component.uuid())) {
+    if (!userSession.hasComponentPermission(UserRole.ADMIN, component) &&
```

Comment:
in fact usually the pattern is to check project permission then to fallback on global permission. In that case `checkComponentAnyPermission()` can't mix project and global permissions in its parameters. An option is to restrict this method to administration permission: `checkComponentAdminPermission(ComponentDto)`.